Sanitizer common_interface_defs not actually common to all sanitizers

`<sanitizer/msan_interface.h>` includes `<sanitizer/common_interface_defs.h>`, so I assumed that those functions would be available under MSan as well as ASan.  But it seems not to be the case:

```console
$ cat foo.c
#include <sanitizer/msan_interface.h>
#include <stdlib.h>

int main(void) {
        char *buf = malloc(1024);
        __sanitizer_annotate_contiguous_container(buf, buf + 1024, buf + 512, buf + 1024);
        return 0;
}
$ clang -fsanitize=memory foo.c
/usr/bin/ld: /tmp/foo-072909.o: in function `main':
foo.c:(.text+0x171): undefined reference to `__sanitizer_annotate_contiguous_container'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
$ clang --version
clang version 18.1.8
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

--- a/compiler-rt/lib/sanitizer_common/CMakeLists.txt
+++ b/compiler-rt/lib/sanitizer_common/CMakeLists.txt
@@ -4,6 +4,7 @@
 set(SANITIZER_SOURCES_NOTERMINATION
   sanitizer_allocator.cpp
   sanitizer_common.cpp
+  sanitizer_contiguous_container.cpp
   sanitizer_deadlock_detector1.cpp
   sanitizer_deadlock_detector2.cpp
   sanitizer_errno.cpp

--- a/compiler-rt/lib/sanitizer_common/sanitizer_common_interface.inc
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_common_interface.inc
@@ -8,18 +8,18 @@
 // Sanitizer Common interface list.
 //===----------------------------------------------------------------------===//
 INTERFACE_FUNCTION(__sanitizer_acquire_crash_state)
-INTERFACE_FUNCTION(__sanitizer_annotate_contiguous_container)
-INTERFACE_FUNCTION(__sanitizer_annotate_double_ended_contiguous_container)
-INTERFACE_FUNCTION(__sanitizer_copy_contiguous_container_annotations)
-INTERFACE_FUNCTION(__sanitizer_contiguous_container_find_bad_address)
-INTERFACE_FUNCTION(
+INTERFACE_WEAK_FUNCTION(__sanitizer_annotate_contiguous_container)
+INTERFACE_WEAK_FUNCTION(__sanitizer_annotate_double_ended_contiguous_container)
+INTERFACE_WEAK_FUNCTION(__sanitizer_copy_contiguous_container_annotations)
+INTERFACE_WEAK_FUNCTION(__sanitizer_contiguous_container_find_bad_address)
+INTERFACE_WEAK_FUNCTION(
     __sanitizer_double_ended_contiguous_container_find_bad_address)
 INTERFACE_FUNCTION(__sanitizer_set_death_callback)
 INTERFACE_FUNCTION(__sanitizer_set_report_path)
 INTERFACE_FUNCTION(__sanitizer_set_report_fd)
 INTERFACE_FUNCTION(__sanitizer_get_report_path)
-INTERFACE_FUNCTION(__sanitizer_verify_contiguous_container)
-INTERFACE_FUNCTION(__sanitizer_verify_double_ended_contiguous_container)
+INTERFACE_WEAK_FUNCTION(__sanitizer_verify_contiguous_container)
+INTERFACE_WEAK_FUNCTION(__sanitizer_verify_double_ended_contiguous_container)
 INTERFACE_WEAK_FUNCTION(__sanitizer_on_print)
 INTERFACE_WEAK_FUNCTION(__sanitizer_report_error_summary)
 INTERFACE_WEAK_FUNCTION(__sanitizer_sandbox_on_notify)

--- a/compiler-rt/lib/sanitizer_common/sanitizer_contiguous_container.cpp
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_contiguous_container.cpp
@@ -1,0 +1,53 @@
+//===-- sanitizer_contiguous_container.cpp
+//-----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+//
+// This file provides weak defs of __sanitizer*contiguous_container* functions
+// whose strong implementations can be defined in particular runtime libs
+// of sanitizers
+//
+//===---------------------------------------------------------------------===//
+
+#include "sanitizer_internal_defs.h"
+
+SANITIZER_INTERFACE_WEAK_DEF(void, __sanitizer_annotate_contiguous_container,
+                             const void *, const void *, const void *,
+                             const void *) {}
+
+SANITIZER_INTERFACE_WEAK_DEF(
+    void, __sanitizer_annotate_double_ended_contiguous_container, const void *,
+    const void *, const void *, const void *, const void *, const void *) {}
+
+SANITIZER_INTERFACE_WEAK_DEF(void,
+                             __sanitizer_copy_contiguous_container_annotations,
+                             const void *, const void *, const void *,
+                             const void *) {}
+
+SANITIZER_INTERFACE_WEAK_DEF(int, __sanitizer_verify_contiguous_container,
+                             const void *, const void *, const void *) {
+  return 0;
+}
+
+SANITIZER_INTERFACE_WEAK_DEF(
+    int, __sanitizer_verify_double_ended_contiguous_container, const void *,
+    const void *, const void *, const void *) {
+  return 0;
+}
+
+SANITIZER_INTERFACE_WEAK_DEF(const void *,
+                             __sanitizer_contiguous_container_find_bad_address,
+                             const void *, const void *, const void *) {
+  return nullptr;
+}
+
+SANITIZER_INTERFACE_WEAK_DEF(
+    const void *,
+    __sanitizer_double_ended_contiguous_container_find_bad_address,
+    const void *, const void *, const void *, const void *) {
+  return nullptr;
+}

